### PR TITLE
Fix disallow-unsafe-type Akka.NET settings and harden unsafe type detection

### PR DESCRIPTION
**Pull request by Arkatufus**
Fixes https://github.com/akkadotnet/akka.net/issues/5740

### CODE REVIEW

File: src/Hyperion/Extensions/TypeEx.cs [hunk 1]
```diff
@@ -192,12 +197,25 @@ private static bool UnsafeInheritanceCheck(Type type)
...
+        public static bool IsDisallowedType<TType>()
+            => IsDisallowedType(typeof(TType));
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsDisallowedType(Type type)
+            => IsDisallowedType(type.FullName);
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsDisallowedType(string name)
+            => UnsafeTypesDenySet.Any(name.Contains);
```
Comment (Arkatufus) [inline src/Hyperion/Extensions/TypeEx.cs:211]: Refactored this out of the method so we can use it in unit tests

File: src/Hyperion/Extensions/TypeEx.cs [hunk 2]
```diff
@@ -208,16 +226,16 @@ public static Type LoadTypeByName(string name, bool disallowUnsafeTypes, ITypeFi
                 // i.e. if there are different version available in GAC and locally
                 var typename = ToQualifiedAssemblyName(name, ignoreAssemblyVersion: false);
                 var type = Type.GetType(typename, true);
-                if (UnsafeInheritanceCheck(type))
+                if (disallowUnsafeTypes && UnsafeInheritanceCheck(type))
```
Comment (Arkatufus) [inline src/Hyperion/Extensions/TypeEx.cs:229]: This is the bug, disallowed types are checked several times, this line need to be bypassed if unsafe types are allowed

File: src/Hyperion/Extensions/TypeEx.cs [hunk 3]
```diff
@@ -208,16 +226,16 @@ public static Type LoadTypeByName(string name, bool disallowUnsafeTypes, ITypeFi
...
                         "Unsafe Type Deserialization Detected!", name);
                 return type;
             }
             catch (IOException)
             {
                 var typename = ToQualifiedAssemblyName(name, ignoreAssemblyVersion: true);
-                var type =  Type.GetType(typename, true);
-                if (UnsafeInheritanceCheck(type))
+                var type = Type.GetType(typename, true);
+                if (disallowUnsafeTypes && UnsafeInheritanceCheck(type))
```
Comment (Arkatufus) [inline src/Hyperion/Extensions/TypeEx.cs:238]: This is the bug, disallowed types are checked several times, this line need to be bypassed if unsafe types are allowed

File: src/Hyperion.Tests/UnsafeDeserializationExclusionTests.cs [hunk 4]
```diff
@@ -25,6 +34,73 @@ public void CantDeserializeANaughtyType()
             }
         }
 
+        [Theory]
+        [MemberData(nameof(DangerousObjectFactory))]
+        public void DetectNaughtyTypesByDefault(Type dangerousType)
```
Comment (Arkatufus) [inline src/Hyperion.Tests/UnsafeDeserializationExclusionTests.cs:39]: Detection hardening, actually try to use the dangerous types and see if they are being caught by the default filter

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**